### DISCUSSION TITLE
Dynamic url for hx-get

### DISCUSSION THREAD

**Discussion by softdevca** [bigskysoftware/htmx#2423] [Q&A]
Is there a way to specify a dynamic URL for `hx-get`?  I would ideally like to do something like `hx-get="js:route(date)"`.

The URL is determined by date and as we all know, dates and time zones are a pain in the ass.  The current date on the server might not be the same on the client depending on time of day and time zones.  Therefor the client needs to specify the URL but it appears `hx-get` only support static URLS.

**Comment by andryyy** [comment]
Isn’t the whole point of query parameters to send dynamic content? :)

You can, as you know, configure them dynamically.

**Comment by softdevca** [comment]
My appologies, I should clarify.  The route doesn't use query parameters, it uses a path, such as `/daily/2024/3/21` which means the usual `hx-include` style techniques won't apply.

**Comment by dgw2jr** [comment]
Your server could determine the clients date with a little extra information:

`
$('#anInputInMyForm').replaceWith(``<input type=hidden id="timeZone" name="timeZone" value="${Intl.DateTimeFormat().resolvedOptions().timeZone}"/>``);
`

Throw an hx-include="#timeZone" on your element that has the hx-get and it should request /daily/2024/3/21?timeZone=America/Chicago

**Reply by softdevca** [reply]
This header would not be available on first page load.  In addition, knowing the time zone will not help because the formatting of the date is locale-specific and will be different on the client and server.

**Reply by andryyy** [reply]
Well, the first problem will exist with a dynamic URL in hx-get, too. Your very first request needs to add that information independently from HTMX. 

If your users need to login, you can set the TZ in a session variable in the backend and make it configurable.

You can also hook on configRequest and add a header for each request containing desired locale information. But that’s just another way to implement it. Using this approach will also allow to change the path. I like @dgw2jr's approach better. It’s more transparent.

**Comment by softdevca** [comment]
Thank you for the suggestion to use `configRequest` but the server does not need to know the users time zone.  Trying to do anything related to time zones not running on the client is a recipe for disaster.  For example, there are cities that have elected out of daylight savings time so a date calculated on the server can be wrong even with the time zone.  Just never do anything related to a local date on the server - not ever.  

The original problem still stands, I have a URL that uses a path instead of query parameters.  Simple enough, except that `hx-get`, `hx-post`, etc.  only take a static URL parameter.  I could use JavaScript to add an `hx-get` parameter and then call `htmx.process()` after load but that seems like a hack.

**Reply by PHvL** [reply]
This issue is quite old, but if others, too, find this issue searching for a solution: see extension `path-params` at https://github.com/bigskysoftware/htmx-extensions/blob/main/src/path-params/README.md

**Reply by MichaelWest22** [reply]
Also a very simple hx-on or event listener on configRequest can solve this issue as the event.detail.path can be modified here just before the request goes out dynamically
